python/feldera/rest/client.py

```diff
@@ -0,0 +1,455 @@
+                )
+                for con in attached_connectors
+            ] if attached_connectors else []
+        )
+
+    def pipelines(self) -> list[Pipeline]:
+        """
+        Get all pipelines
+        """
+
+        resp = self.http.get(
+            path="/pipelines",
+        )
+
+        return [self.__pipeline_from_dict(pipeline) for pipeline in resp]
+
+    def get_pipeline(self, name: str) -> Pipeline:
+        """
+        Get a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}",
+        )
+
+        return self.__pipeline_from_dict(resp)
+
+    def create_pipeline(self, pipeline: Pipeline):
+        """
+        Create a pipeline
+        """
+
+        body = {
+            "config": pipeline.config,
+            "description": pipeline.description or "",
+            "connectors": pipeline.attached_connectors,
+            "program_name": pipeline.program_name,
+        }
+
+        resp = self.http.put(
+            path=f"/pipelines/{pipeline.name}",
+            body=body,
+        )
+
+        pipeline.id = resp.get("pipeline_id")
+
+    def get_pipeline_config(self, name: str) -> dict:
+        """
+        Get the configuration of a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/config",
+        )
+
+        return resp
+
+    def validate_pipeline(self, name: str) -> bool:
+        """
+        Validate a pipeline.
+        Checks whether the pipeline is configured correctly.
+        This includes checking whether the pipeline references a valid compiled program,
+        whether the connectors reference valid tables/views in the program, and more.
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/validate",
+        )
+
+        return "success" in resp
+
+    def delete_pipeline(self, name: str):
+        """
+        Deletes a pipeline by name
+        :param name: The name of the pipeline
+        """
+        resp = self.http.delete(
+            path=f"/pipelines/{name}",
+        )
+
+    def get_pipeline_stats(self, name: str) -> dict:
+        """
+        Get the pipeline metrics and performance counters
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/stats",
+        )
+
+        return resp
+
+    def start_pipeline(self, pipeline_name: str):
+        """
+        Start a pipeline
+        :param pipeline_name: The name of the pipeline to start
+        """
+        self.http.post(
+            path=f"/pipelines/{pipeline_name}/start",
+        )
+
+        while True:
+            status = self.get_pipeline(pipeline_name).state.get("current_status")
+
+            if status == "Running":
+                break
+            elif status == "Failed":
+                raise RuntimeError(f"Failed to start pipeline")
+
+            logging.debug("still starting %s, waiting for 100 more milliseconds", pipeline_name)

```

Comment:
Might be useful to add how long compilation has lasted already: "Starting pipeline %s (so far took %d ms)..."